### PR TITLE
Added the script for lambda invocation

### PR DESCRIPTION
**Pull request by aditya-passbird**
Link to the documentation tutorial : https://metakeep-dev.readme.io/reference/getting-started-with-your-api

### CODE REVIEW

File: lambda/voting/scripts/utils.mjs [hunk 1]
```diff
@@ -0,0 +1,54 @@
+//Helper functions for the MetaKeep APIs
+
+import fetch from "node-fetch";
+
+export default async function getDeveloperWallet() {
```
Comment (rohan-passbird) [inline lambda/voting/scripts/utils.mjs:5]: copy this again. I had added result.ok check.

Same for the transaction status function below

File: lambda/voting/scripts/invoke.mjs [hunk 2]
```diff
@@ -0,0 +1,74 @@
+import fetch from "node-fetch";
+import fs from "fs";
+import env from "dotenv";
+import { exit } from "process";
+
+import getDeveloperWallet, {
```
Comment (rohan-passbird) [inline lambda/voting/scripts/invoke.mjs:6]: rename this file to vote.mjs

File: lambda/voting/scripts/invoke.mjs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  const url = "https://api.metakeep.xyz/v2/app/lambda/invoke";
+
+  const headers = {
+    Accept: "application/json",
+    "Content-Type": "application/json",
+    "x-api-key": process.env.API_KEY,
+    "Idempotency-Key": "Idempotency-Key" + Math.floor(Math.random() * 10000),
+  };
+
+  const data = JSON.parse(
```
Comment (rohan-passbird) [inline lambda/voting/scripts/invoke.mjs:27]: is this needed?

File: lambda/voting/.env [hunk 4]
```diff
@@ -1,3 +1,3 @@
 API_KEY = 
-
+LAMBDA_ADDRESS =
```
Comment (rohan-passbird) [inline lambda/voting/.env:2]:
```suggestion

// Your lambda contract address.
// You get this after running the create script.
LAMBDA_ADDRESS = 
```

File: lambda/voting/scripts/utils.mjs (hunk 1 above)
Comment (aditya-passbird) [reply lambda/voting/scripts/utils.mjs:5]: done

File: lambda/voting/scripts/invoke.mjs (hunk 2 above)
Comment (aditya-passbird) [reply lambda/voting/scripts/invoke.mjs:6]: done

File: lambda/voting/scripts/invoke.mjs (hunk 3 above)
Comment (aditya-passbird) [reply lambda/voting/scripts/invoke.mjs:27]: done

File: lambda/voting/README.md [hunk 5]
```diff
@@ -42,4 +42,4 @@ If you need help with this project, or with MetaKeep APIs in general, please rea
 
 For the latest news about MetaKeep, [follow us on Twitter](https://twitter.com/metakeep), and don't forget to star [our GitHub repository](https://github.com/PassbirdCo/MetaKeepCodeSamples.git)!
 
-**Happy_building_!**
+**Happy*building*!**
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/README.md:51]: done

File: lambda/voting/scripts/registerAndVote.mjs [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+    body: JSON.stringify(requestBody),
+  };
+
+  const result = await fetch(url, options);
+  const resultJson = await result.json();
+  console.log(resultJson);
+  return resultJson;
+}
+
+async function waitUntilTransactionMined(resultJson) {
```
Comment (rohan-passbird) [inline lambda/voting/scripts/registerAndVote.mjs:42]: move this to utils too

File: lambda/voting/scripts/registerAndVote.mjs [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+  // Waits for the transaction to be mined.
+  await waitUntilTransactionMined(resultJson);
+  console.log(
+    "Lambda invocation for registering user completed: " +
+      resultJson.transactionHash +
+      "\n"
+  );
+
+  // Waits for 5 seconds.
+  sleep(5000);
```
Comment (rohan-passbird) [inline lambda/voting/scripts/registerAndVote.mjs:101]: why are we sleeping here?

**Review by rohan-passbird — DISMISSED** [review]
👍

File: lambda/voting/scripts/registerAndVote.mjs [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+  const options = {
+    method: "POST",
+    headers: headers,
+    body: JSON.stringify(requestBody),
+  };
+
+  const result = await fetch(url, options);
+  const resultJson = await result.json();
+  console.log(resultJson);
+  return resultJson;
```
Comment (rohan-passbird) [inline lambda/voting/scripts/registerAndVote.mjs:39]: exit if result not OK just like other functions

File: lambda/voting/scripts/registerAndVote.mjs [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+}
+
+async function main() {
+  env.config();
+
+  // Checks if the API_KEY is set in the .env file.
+  checkAPIKey();
+
+  // Gets the user Address to register as Candidate in Voting contract.
+  const userAddress = await getUserWallet("adityadhir97@gmail.com");
```
[the remark this answers is not in the document]
Comment (aditya-passbird) [reply lambda/voting/scripts/registerAndVote.mjs:81]: done

File: lambda/voting/scripts/registerAndVote.mjs (hunk 7 above)
Comment (aditya-passbird) [reply lambda/voting/scripts/registerAndVote.mjs:101]: to give pause between registerCandidate call and VoteForCandidate call

File: lambda/voting/scripts/utils.mjs [hunk 10]
```diff
@@ -101,6 +101,38 @@ export const getTransactionStatus = async (transactionId) => {
   return resultJson;
 };
 
+// Waits for the transaction to be mined.
+export const waitUntilTransactionMined = async (resultJson) => {
```
Comment (rohan-passbird) [inline lambda/voting/scripts/utils.mjs:105]: update other files to use this function too.

File: lambda/voting/scripts/registerAndVote.mjs (hunk 8 above)
Comment (aditya-passbird) [reply lambda/voting/scripts/registerAndVote.mjs]: done

**Review by rohan-passbird — APPROVED** [review]
👍

**Pull request merged**